logo.js

```diff
@@ -1601,13 +1740,37 @@ function LogoInterpreter(turtle, stream, savehook)
   def("false", function() { return 0; });
 
   def("and", function(a, b) {
-    return Array.from(arguments).every(function(f) { return f(); }) ? 1 : 0;
+    var args = Array.from(arguments);
+    return checker(args, function (value) {return ! value;});
   }, {noeval: true});
 
   def("or", function(a, b) {
-    return Array.from(arguments).some(function(f) { return f(); }) ? 1 : 0;
+    var args = Array.from(arguments);
+    return checker(args, function (value) {return value;});
   }, {noeval: true});
 
+  function _checker(args, shouldStop) {
+    if (! args.length) {
+      return Promise.resolve(true);
```

Comment:
Might be an existing bug, but what should `(AND)` and `(OR)` (i.e. called with 0 args) evaluate to?